template/src/navigation/types.ts

```diff
@@ -0,0 +1,47 @@
+import {BottomTabScreenProps} from '@react-navigation/bottom-tabs'
+import {CompositeScreenProps} from '@react-navigation/native'
+import {StackScreenProps} from '@react-navigation/stack'
+
+export interface INavigationParams {
+  params?: INavigationOnlyParams | INavigationScreenAndParams
+}
+
+export type INavigationOnlyParams = object
+export type INavigationScreenAndParams = {
+  screen?: string
+  params?: INavigationOnlyParams | INavigationScreenAndParams
+}
+
+export type HomeStackParamList = {
+  HomeScreen: undefined
+}
+
+export type AuthStackParamList = {
+  LoginScreen: undefined
+  SignUpScreen: undefined
```

Comment:
Why `undefined`, maybe we should use `unknown`

Reply:
unknown doesn't satisfy navigation props which is undefined | object

Reply:
Ok cleared